dds/DCPS/RemoveAssociationSweeper.h

```diff
@@ -203,33 +218,56 @@ void RemoveAssociationSweeper<T>::ScheduleCommand::execute()
    * Pass pointer to writer info for timer to use, must decrease ref count when
    * canceling timer
    */
-  const void* arg = reinterpret_cast<const void*>(this->info_.in());
-  this->sweeper_->info_set_.push_back(this->info_);
+  RcHandle<RemoveAssociationSweeper<T> > sweeper = this->sweeper_.lock();
+  if (!sweeper) {
+    return;
+  }
+
+  RcHandle<WriterInfo>& info = this->info_;
+
+  const void* arg = new TimerArg(sweeper, info);
```

Comment:
Could this move down to right before it's used?  Or even eliminate the need for a distinct local variable?